Bottom: c5adac54320e209cd41c2c34983c5eaf0fc61b71
Top:    c21e6e664d5aafd0cb11b9d2729a709f4f28045b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-01-27 21:37:55 +0000

Cure crosshair lagging on screen

FIXME: SHOULD NOT RE-WRITE THE GTK HID


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 3b543cb..c838117 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -671,11 +671,12 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   static gint x_prev = -1, y_prev = -1;
   gboolean moved;
 
-
   gdk_event_request_motions (ev);
 
+#if 0
   if (!ghid_start_drawing (out))
     return FALSE;
+#endif
 
 
   if (out->panning)
@@ -695,11 +696,14 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   queue_tooltip_update (out);
 #endif
 
-  ghid_show_crosshair (TRUE);
-  if (moved && have_crosshair_attachments ())
+//  ghid_show_crosshair (TRUE);
+  if (moved) // && have_crosshair_attachments ())
     ghid_draw_area_update (gport, NULL);
 
+#if 0
   ghid_end_drawing (out);
+#endif
+
   return FALSE;
 }
